features/cellular/framework/API/CellularDevice.h

```diff
@@ -132,28 +225,46 @@ class CellularDevice {
      */
     virtual void modem_debug_on(bool on) = 0;
 
-    /** Get network stack.
-     *
-     *  @return network stack
-     */
-    virtual NetworkStack *get_stack() = 0;
-
     /** Initialize cellular module must be called right after module is ready.
      *  For example, when multiple modules are supported in a single AT driver this function detects
      *  and adapts to an actual module at runtime.
      *
-     *  @param fh    file handle used in communication to modem.
-     *
      *  @return 0 on success
      */
-    virtual nsapi_error_t init_module(FileHandle *fh) = 0;
+    virtual nsapi_error_t init_module() = 0;
+
+    /** Get the linked list of CellularContext instances
+     *
+     *  @return Pointer to first item in linked list
+     */
+    virtual CellularContext *get_context_list() const;
 
 protected:
+    friend class AT_CellularNetwork;
+    friend class AT_CellularContext;
+    /** Cellular callback which is called by Network class after this class attaches to it, CellularStateMachine
+     *  and CellularContext when in PPP mode. This method will broadcast to every interested classes:
+     *  CellularContext (might be many) and CellularStateMachine if available.
+     *
+     */
+    void cellular_callback(nsapi_event_t ev, intptr_t ptr);
```

Comment:
Could this be rephrased more clear? It is not clear who is actually calling it and to which case PPP reffers to.

Reply:
fixed.